rhel_osv/csaf.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2024 Jason Shepherd
+# License: GPLv3+
+"""Red Hat CSAF parser"""
+import json
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Iterable
+
+logger = logging.Logger(__name__)
+
+
+class Remediation:
+    """
+    class to handle remediation advice in CSAF data
+    """
+
+    # pylint: disable=too-few-public-methods
+    # This class is used for initialization and encapsulation of Remediation data
+
+    def __init__(self, csaf_product_id: str, cpes: dict[str, str],
+                 purls: dict[str, str]):
+        if ":" not in csaf_product_id:
+            raise ValueError(
+                f"Did not find ':' in product_id: {csaf_product_id}")
+        (self.product,
+         self.product_version) = csaf_product_id.split(":", maxsplit=1)
+        self.cpe = cpes.get(self.product)
+        self.purl = purls.get(self.product_version)
+        # There are many pkg:oci/ remediations in Red Hat data. However there are no strict
+        # rules enforced on versioning Red Hat containers, therefore we cant compare container
+        # versions to each other with 100% accuracy at this time.
+        if not self.purl.startswith("pkg:rpm/"):
+            raise ValueError(
+                "Non RPM remediations are not supported in OSV at this time")
+
+        # NEVRA stands for Name Epoch Version Release and Architecture
+        # We split the name from the rest of the 'version' data (EVRA). We store name as component.
+        split_component_version = self.product_version.rsplit("-", maxsplit=2)
+        if len(split_component_version) != 3:
+            raise ValueError(
+                f"Could not convert component into NEVRA: {self.product_version}"
+            )
+        self.component = split_component_version[0]
+        self.fixed_version = "-".join(
+            (split_component_version[1], split_component_version[2]))
+
+
+class Vulnerability:
+    """
+    class to handle vulnerability information
+    """
+
+    # pylint: disable=too-few-public-methods
+    # This class encapsulates Red Hat CSAF Vulnerability data
+    # Only initialization is required because data retrival is via JSON encoding
```

Comment:
Fix spelling mistake

```suggestion
    # Only initialization is required because data retrieval is via JSON encoding
```

Reply:
Oops, thanks.